spec/requests/groups/membership_spec.rb

```diff
@@ -1,117 +1,132 @@
-        expect(flash[:notice]).to eq(
-          "You have removed #{another_user.name} from #{group.name}"
-        )
-      end
-    end
-
-    context 'when current_user is not a leader' do
-      let!(:group_member) { create :group_member, user_id: controller.current_user.id, leader: false }
-      let!(:group) { group_member.group }
-      let!(:another_user) { create :user2 }
-      let!(:another_member) { create :group_member, user_id: another_user.id, leader: false, group: group }
-
-      it 'cannot kick another group member out' do
-        delete :kick, params: {
-          group_id: group.id,
-          member_id: another_user.id
-        }
-        expect(response).to redirect_to(groups_path)
-        expect(flash[:alert]).to eq(
-          'You cannot kick a group member because you are not a leader.'
-        )
-      end
-    end
-  end
-
-  describe '#leave' do
-    context 'when current_user is not a leader' do
-      let!(:leader_user) { create :user2 }
-      let!(:leader_member) { create :group_member, user_id: leader_user.id, leader: true }
-      let!(:another_member) { create :group_member, user_id: controller.current_user.id, leader: false }
-      let!(:group) { another_member.group }
-
-      it 'can leave group' do
-        delete :destroy, params: { group_id: group.id }
-        expect(response).to redirect_to(groups_path)
-        expect(flash[:notice]).to eq(
-          "You have left #{group.name}"
-        )
-      end
-    end
-
-    context 'when current_user is a leader' do
-      let!(:group_member) { create :group_member, user_id: controller.current_user.id, leader: true }
-
-      it 'can leave the group when user is not the only leader' do
-        group = group_member.group
-        another_user = create :user2
-        another_member = create :group_member, user_id: another_user.id, leader: true, group: group
-        delete :destroy, params: { group_id: group.id }
-        expect(response).to redirect_to(groups_path)
-        expect(flash[:notice]).to eq(
-          "You have left #{group.name}"
-        )
-      end
-
-      it 'cannot leave the group when user is the only leader' do
-        delete :destroy, params: { group_id: group_member.group.id }
-        expect(response).to redirect_to(groups_path)
-        expect(flash[:alert]).to eq(
-          'You cannot leave the group, you are the only leader.'
-        )
-      end
-    end
-  end
-end
+# frozen_string_literal: true
+
+RSpec.describe 'Membership', type: :request do
+  before { sign_in current_user }
+
+  let(:current_user) { create(:user1) }
+
+  describe '#create' do
+    context 'new group member for existing group' do
+      it 'adds new member' do
+        # Arrange
+        leader_user = create :user2
+        group_leader = create :group_member, user_id: leader_user.id, leader: true
+        group = group_leader.group
+
+        # Act
+        post group_membership_path(group_id: group.id)

```

Comment:
Normally for most request specs I've wrote, I could've just done something like the following:
```suggestion
        post group_membership_path(group)
```

Rails magic would usually assume I'm passing the `group.id` to `:id` for that specified route. 

For some reason I'll get a failed test with the following instead:

```
Expected response to be a redirect to <http://www.example.com/groups/test-group> but was a redirect to <http://www.example.com/groups>.
       Expected "http://www.example.com/groups/test-group" to be === "http://www.example.com/groups".
```

For the other tests, if I just pass the model, I might get 500 error. I'm assuming this could depend on the `route.rb` setup, but base on the current setup, I can define the route's path variables and pass the corresponding `id` to the paths I'd like it to go to unless I'm missing something.

Reply:
Ah yeah the reason for this is cause we're expecting `group_id` instead of `id` like you pointed out. We could refactor [controller](https://github.com/ifmeorg/ifme/blob/master/app/controllers/groups/memberships_controller.rb) and the corresponding route so we're expecting an `id`. I don't think this is a pressing thing to fix to but we could make a note of it. Thanks for pointing this out!